arches_component_lab/src/arches_component_lab/widgets/TextWidget/components/TextWidgetEditor.vue

```diff
@@ -0,0 +1,104 @@
+<script setup lang="ts">
+import { ref, watch, watchEffect } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import InputText from "primevue/inputtext";
+import Select from "primevue/select";
+
+import { fetchLanguages } from "@/arches_component_lab/widgets/api.ts";
+import GenericFormField from "@/arches_component_lab/generics/GenericFormField.vue";
+
+import type { FormFieldResolverOptions } from "@primevue/forms";
+import type { Language } from "@/arches_component_lab/types.ts";
+import type { StringValue } from "@/arches_component_lab/datatypes/string/types.ts";
+
+const props = defineProps<{
+    nodeAlias: string;
+    value: StringValue;
+}>();
+
+const languages = ref<Language[]>([]);
+const selectedLanguage = ref<Language>();
+const managedNodeValue = ref<StringValue["node_value"]>();
+const singleInputValue = ref<string>();
+
+const { $gettext } = useGettext();
+
+watchEffect(async () => {
+    languages.value = (await fetchLanguages()) as Language[];
+    selectedLanguage.value = languages.value.find((lang) => lang.isdefault);
+});
+
+watch(languages, () => {
+    const workingObject = { ...props.value.node_value };
+    for (const language of languages.value) {
+        if (!workingObject[language.code]) {
+            workingObject[language.code] = {
+                value: "",
+                direction: language.default_direction,
+            };
+        }
+    }
+    managedNodeValue.value = workingObject;
+});
+
+watch(selectedLanguage, () => {
+    if (selectedLanguage.value && managedNodeValue.value) {
+        singleInputValue.value =
+            managedNodeValue.value[selectedLanguage.value.code].value;
+    } else {
+        singleInputValue.value = "";
+    }
+});
```

Comment:
Now, unlike the core resource editor, when you switch languages we clear the entry for that language. This implementation is more straightforward. Having values in the background that you can't see that get saved seems harder to implement, so I'm waiting for guidance on that point before implementing. Don't want to "build old bugs into the new system".

Reply:
oh fair, I don't have a strong opinion here ( surprisingly lol ) but I'm sure others do. IMO this is fine for an MVP 👍